### PR TITLE
docs: Oryx compositions

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
closes [HRZ-2158](https://spryker.atlassian.net/browse/HRZ-2158)

[HRZ-2158]: https://spryker.atlassian.net/browse/HRZ-2158?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+title: "Oryx: Compositions"
+description: Compositions are used to organize components and provide layout in Oryx applications
+last_updated: June 8, 2023
+template: concept-topic-template
+---
+
+Compositions in Oryx are a powerful tool for organizing components and defining their layout. The removes page specific layout concerns from the component implementation, which makes the components less opinionated and therefor better reusable.
```
Comment (tolerants) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:8]:
```suggestion
Compositions in Oryx are a powerful tool for organizing components and defining their layout. It removes page specific layout concerns from the component implementation, which makes the components less opinionated and therefore better reusable.
```

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+## Overview
+
+A composition contains a list of components which are rendered in the DOM in their specific order. A composition itself is a component, which means that you can easily created nested composition structures. In fact, a page itself is a composition.
```
Comment (tolerants) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:12]:
```suggestion
A composition contains a list of components which are rendered in the DOM in their specific order. A composition itself is a component, which means that you can easily create nested composition structures. In fact, a page itself is a composition.
```

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+Compositions in Oryx take a data-driven approach, allowing you to configure the composition using external data sources. This approach offers several potential benefits:
+
+- **Avoid Hard-Coded Page Layout**: By using data to configure compositions, you can avoid hard-coding the page layout in your application code. Instead, you can define the structure and layout of your pages using external data, making it more flexible and easier to customize.
```
Comment (tolerants) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:16]: Maybe we can also mention that we are separating application/component code from the layout and as a result the customer will benefit from it during upgrades.

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    }
+  </style>
+</oryx-composition>
+
+When the data structure does not provide an unique id, a `uid` will be
+autogenerated. The example demonstrates the need to have a uid, as the data
+driven styles are bound to elements with those uids. Nested compositions are
+added to the DOM as you would expect, they are just placed as part of the list
+of inner components of a composition. The following example shows the components
+of the cart page: ```html
```
Comment (tolerants) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:63]:
```suggestion
of the cart page: 

```html
```

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    :host([uid="static130"]),
+    [uid="static130"] {
+      width: 50%;
+      margin: auto;
+    }
+  </style>
+</oryx-composition>
+
+When the data structure does not provide an unique id, a `uid` will be
+autogenerated. The example demonstrates the need to have a uid, as the data
```
Comment (tolerants) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:59]:
```suggestion
autogenerated. The example demonstrates the need to have a `uid`, as the data
```

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+Compositions can be nested. Every time you need to put in additional structure for the organization and layout of a section, you can bring in a nested composition. A good example of nested compositions is the footer of an e-commerce storefront. The footer typically comes with a lot of links. The links are organized in groups, to better group them by corporate links, self service links, legal links, social link, etc. For a good experience, the groups have a layout assigned to them, to put them on the left or right, top of bottom of the footer.
+
+## Generated HTML
+
+The generated HTML represents the composition hierarchy and defines the rendering order of components within Oryx applications. Understanding the generated DOM structure is important for working effectively with compositions.
+
+The DOM structure is based on the experience data provided for each composition. It reflects the composition hierarchy and the components included within the composition. Let's take the login page composition as an example. Based on the composition data, which only holds a single component, the following html is generated:
+
+````html
```
Comment (tolerants) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:46]: Let's add a composition and then a result of generation maybe.

**Review by tolerants — CHANGES_REQUESTED** [review]

**Review by tolerants — DISMISSED** [review]

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md [hunk 7]
```diff
@@ -37,11 +37,17 @@ export const b2cFeatures: AppFeature[] = [
...
+
 For more information about feature sets, see [Feature sets](/docs/scos/dev/front-end-development/{{page.version}}/oryx/oryx-feature-sets.html)
 
 ## Themes
 
-The overarching UI of the application is driven by themes. To get started with a default theme, you can import it from the preset package and apply it to your Oryx application:
+Themes in Oryx play a key role in defining the visual appearance of your application. A theme represents the global typography, colors, and other specific design elements, such as form field placeholder color, that define the overall look and feel of your application.
+
+The theme consists of design tokens, which are configuration values that control the visual aspects of your application. Design tokens allow you to define global typography, colors, spacing, and other visual properties. By configuring these design tokens, you can customize the visual appearance of your application to align with your brand or specific design requirements.
```
Comment (dunqan) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md:48]: In those two paragraphs we are mentioning almost the same thing three times:
- "that define the overall look and feel of your application"
- "that control the visual aspects of your application"
- "customize the visual appearance of your application ..."

Not saying it's necesairly bad, but a bit noisy IMO.

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md [hunk 8]
```diff
@@ -53,7 +59,11 @@ export const app = appBuilder()
   .create();
 ```
 
-The theme contains mainly design tokens that are used inside the components styles.
+By utilizing themes and design tokens in Oryx presets, you have the flexibility to customize the visual aspects of your application, making it unique and aligned with your branding or specific design preferences.
```
Comment (dunqan) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md:62]: another one
- "customize the visual aspects of your application"

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md [hunk 9]
```diff
@@ -53,7 +59,11 @@ export const app = appBuilder()
...
+
+Themes play a crucial role in maintaining a consistent and coherent visual experience throughout your application. They provide a centralized way to manage and apply design tokens, ensuring a unified look and feel across components and screens.
+
+Take advantage of the themes provided in Oryx presets or create your own themes to tailor the visual appearance of your application according to your specific needs.
```
Comment (dunqan) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md:66]: one more :) 
- ailor the visual appearance of your application

**Review by dunqan — APPROVED** [review]

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+The data configuration for a composition typically includes information about the components, their order, layout rules, styling options, and any other properties relevant to the composition.
+
+By separating the data configuration from the code, you can easily update and manage the composition without modifying the underlying application code. This promotes a more efficient development process and enables non-technical stakeholders to contribute to the composition customization.
+
+## Nested compositions
+
+Compositions can be nested. Every time you need to put in additional structure for the organization and layout of a section, you can bring in a nested composition. A good example of nested compositions is the footer of an e-commerce storefront. The footer typically comes with a lot of links. The links are organized in groups, to better group them by corporate links, self service links, legal links, social link, etc. For a good experience, the groups have a layout assigned to them, to put them on the left or right, top of bottom of the footer.
+
+## Generated HTML
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:42]: Please add more context to the title. How is it related to composition?

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+      margin: auto;
+    }
+  </style>
+</oryx-composition>
+```
+
+When the data structure does not provide an unique id, a `uid` will be
+autogenerated. The example demonstrates the need to have a `uid`, as the data
+driven styles are bound to elements with those uids. Nested compositions are
+added to the DOM as you would expect, they are just placed as part of the list
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:84]: Rephrase the sentence to remove `as you would expect`.

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-feature-sets.md [hunk 12]
```diff
@@ -26,6 +26,67 @@ Oryx includes predefined feature sets that cover common use cases for web applic
 - b2cFeatures: features commonly used in B2C applications.
 - fulfillmentFeatures: features used in PWAs, used for picking products for fulfillment.
 
+## Static Experience Data
+
+Feature sets also contain (static) experience data. Experience data includes the structure and layout of the components of the application. Feature sets provide all the static pages data that is required for the application experiences. The `b2cFeatures` feature set, for example, comes with the following static data:
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-feature-sets.md:31]: `Feature sets provide all the static pages data that is required for the application experiences.`

What new information does this sentence add?

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md [hunk 13]
```diff
@@ -37,11 +37,15 @@ export const b2cFeatures: AppFeature[] = [
...
 ```
 
+Feature sets also contain static experience data. Experience data includes the structure and layout of the components of the application, such as pages and sections. By utilizing the static experience data provided by the presets, you do no need to set up any boilerplate code. Moreover, we avoid hardcoded page structures that would not allow for personalized experiences going forward.
+
 For more information about feature sets, see [Feature sets](/docs/scos/dev/front-end-development/{{page.version}}/oryx/oryx-feature-sets.html)
 
 ## Themes
 
-The overarching UI of the application is driven by themes. To get started with a default theme, you can import it from the preset package and apply it to your Oryx application:
+A theme represents the global visual appearance of an application, including typography, colors, and other specific design elements, such as form field placeholder color. The theme is build with design tokens, which are configurable values that are used in CSS properties. By configuring these design tokens, you can customize the theme and align the application and it's components with your brand identity or specific design requirements.
```
Comment (andriitserkovnyi) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md:46]: Is this a list? `form field placeholder color`

**Review by jmartinspryker — APPROVED** [review]

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-feature-sets.md (hunk 12 above)
Comment (tobi-or-not-tobi) [reply docs/scos/dev/front-end-development/202212.0/oryx/oryx-feature-sets.md:31]: It's the first time that we talk about "pages". Pages will have more attention in the docs going forward, but it is partially covered by static experience data.

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md (hunk 13 above)
Comment (tobi-or-not-tobi) [reply docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md:46]: i've rephrased it

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md
Comment (jmartinspryker) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:1]: makes the components more versatile for reuse.

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
+---
+title: "Oryx: Compositions"
+description: Compositions are used to organize components and provide layout in Oryx applications
+last_updated: June 8, 2023
+template: concept-topic-template
+---
+
+Compositions in Oryx are a powerful tool for organizing components and defining their layout. It removes page-specific layout concerns from the component implementation, which makes the components less opinionated and more reusable.
```
Comment (jmartinspryker) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:8]: makes the components more versatile for reuse.

Comment (VadymSachenko) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:8]: markteting slang
```suggestion
Compositions in Oryx are a tool for organizing components and defining their layout. It removes page-specific layout concerns from the component implementation, which makes the components less opinionated and more reusable.
```

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+
+## Overview of compositions
+
+A composition contains a list of components which are rendered in the DOM in their specific order. A composition itself is a component, which means that you can easily create nested composition structures. In fact, a page itself is a composition.
```
Comment (VadymSachenko) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:12]:
```suggestion
A composition contains a list of components that are rendered in the DOM in their specific order. A composition itself is a component, which means that you can easily create nested composition structures. In fact, a page itself is a composition.
```

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+
+Compositions take a data-driven approach, letting you configure the composition using external data sources. This approach offers several potential benefits:
+
+- Avoid hard-coded page layout: By using data to configure compositions, you can avoid hardcoding the page layout in your application code. Instead, you can define the structure and layout of your pages using external data, making it more flexible and easier to customize.
+
+- Upgradable composition: A configurable data set is easier to upgrade. Instead of upgrading to a hardcoded component structure, you can select an alternative data set that will hold new components. This makes it easier to _opt in_ to alternative compositions.
+
+- No-code customizations: The data-driven approach enables no-code customizations of the compositions. With the use of a What You See Is What You Get (WYSIWYG) tool, non-technical users can easily modify a composition by adjusting the data configuration, without the need to edit the underlying code.
```
Comment (VadymSachenko) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:20]:
```suggestion
- No-code customizations: The data-driven approach enables no-code customizations of the compositions. With the use of a What You See Is What You Get (WYSIWYG) tool, non-technical users can easily modify a composition by adjusting the data configuration without the need to edit the underlying code.
```

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+
+- Alternative data sets per screen size: Oryx compositions support the configuration of alternative data sets based on screen sizes. You can define different experiences for different devices, such as mobile, tablet, and desktop. By adapting a data set based on screen size, you optimize the user experience.
+
+The data-driven approach in Oryx compositions empowers you to dynamically configure the structure, layout, and content of your application based on external data sources. This flexibility enables easy customization, experimentation, and personalization, leading to enhanced user experiences and improved performance.
+
+## Configuring compositions with data
+
+To utilize the data-driven approach, you can provide the composition data using static-experience data, or by loading the data from an external source like a backend API. For an example of how the static–experience data is provided, see [Feature sets](/docs/scos/dev/front-end-development/{{page.version}}/oryx/oryx-feature-sets.html).
```
Comment (VadymSachenko) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:32]:
```suggestion
To utilize the data-driven approach, you can provide the composition data using static-experience data or by loading the data from an external source like a backend API. For an example of how the static–experience data is provided, see [Feature sets](/docs/scos/dev/front-end-development/{{page.version}}/oryx/oryx-feature-sets.html).
```

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+
+The data configuration for a composition typically includes information about the components: their order, layout rules, styling options, and any other properties relevant to the composition.
+
+By separating the data configuration from the code, you can easily update and manage the composition without modifying the underlying application code. This promotes a more efficient development process and enables non-technical stakeholders to contribute to the composition customization.
+
+## Nested compositions
+
+Compositions can be nested. When you need to put in an additional structure for the organization and layout of a section, you can use a nested composition. A good example of nested compositions is the footer of an e-commerce storefront. The footer typically comes with a lot of links. The links are organized in groups, like self service links, legal, and social. For a good experience, the groups have a layout assigned to them, to put them on the left or right, top of bottom of the footer.
```
Comment (VadymSachenko) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-compositions.md:40]:
```suggestion
Compositions can be nested. When you need to put in an additional structure for the organization and layout of a section, you can use a nested composition. A good example of nested compositions is the footer of an e-commerce storefront. The footer typically comes with a lot of links. The links are organized in groups, like self-service links, legal, and social. For a good experience, the groups have a layout assigned to them, to put them on the left or right, top or bottom of the footer.
```

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md [hunk 19]
```diff
@@ -37,11 +37,15 @@ export const b2cFeatures: AppFeature[] = [
   ...
 ```
 
-For more information about feature sets, see [Feature sets](/docs/scos/dev/front-end-development/{{page.version}}/oryx/oryx-feature-sets.html)
+Feature sets also contain static experience data. Experience data includes the structure and layout of the components of and application, such as pages and sections. By utilizing the static experience data provided by the presets, you don't need to set up any boilerplate code. Moreover, we avoid hardcoded page structures which enables personalized experiences.
```
Comment (VadymSachenko) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md:40]:
```suggestion
Feature sets also contain static experience data. Experience data includes the structure and layout of the components of an application, such as pages and sections. By utilizing the static experience data provided by the presets, you don't need to set up any boilerplate code. Moreover, we avoid hardcoded page structures that enable personalized experiences.
```

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md [hunk 20]
```diff
@@ -37,11 +37,15 @@ export const b2cFeatures: AppFeature[] = [
...
+
+For more information about feature sets, see [Feature sets](/docs/scos/dev/front-end-development/{{page.version}}/oryx/oryx-feature-sets.html).
 
 ## Themes
 
-The overarching UI of the application is driven by themes. To get started with a default theme, you can import it from the preset package and apply it to your Oryx application:
+A *theme* represents the global visual appearance of an application, including typography, colors, and other specific design elements, such as the color of a form field's placeholder. Themes are  built with design tokens, which are configurable values that are used in CSS properties. By configuring these design tokens, you can customize a theme and align the application and it's components with your brand identity or specific design requirements.
```
Comment (VadymSachenko) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md:46]:
```suggestion
A *theme* represents the global visual appearance of an application, including typography, colors, and other specific design elements, such as the color of a form field's placeholder. Themes are built with design tokens, which are configurable values that are used in CSS properties. Configuring these design tokens lets you customize a theme and align the application and its components with your brand identity or specific design requirements.
```

File: docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md [hunk 21]
```diff
@@ -53,7 +57,7 @@ export const app = appBuilder()
   .create();
 ```
 
-The theme contains mainly design tokens that are used inside the components styles.
+Themes play a crucial role in maintaining a consistent and coherent visual experience throughout your application. They provide a centralized way to manage and apply design tokens, ensuring a unified look and feel across components and screens.
```
Comment (VadymSachenko) [inline docs/scos/dev/front-end-development/202212.0/oryx/oryx-presets.md:60]: Looks like marketing slang
```suggestion
Themes help maintain a consistent and coherent visual experience throughout your application. They provide a centralized way to manage and apply design tokens, ensuring a unified look and feel across components and screens.
```

**Review by VadymSachenko — CHANGES_REQUESTED** [review]

**Review by VadymSachenko — APPROVED** [review]

**Review by andriitserkovnyi — APPROVED** [review]

**Pull request merged**